### PR TITLE
Properly handle VB WithEvents fields in metadata code model elements

### PR DESCRIPTION
**Pull request by DustinCampbell**
**Customer Scenario**:

When inheriting a VB WebForm from another VB WebForm that was compiled into a different assembly, the designer will generate duplicate field declarations for inherited controls, producing warnings in inherited WebForm. There is a workaround in that the user can manually delete the duplicated field declarations from the generated code in the *.designer.vb file. However, if the user makes any change to the designer, the field declarations will be re-generated, requiring them to be deleted again.

Filed on Connect [here](https://connect.microsoft.com/VisualStudio/feedback/details/1695089/the-designer-vb-for-inherited-web-forms-are-automatically-filled-up-with-duplicate-declarations).

**Issue Description**:

The issue here is that VB WithEvents fields are not represented properly "external" code model elements -- that is, code model elements created from metadata. The WebForm designer walks all of the inherited fields to determine which fields need to be generated, and VB WithEvents fields are simply get missed because they don't appear in a form that the WebForm designer expects them to.

**Fix Description**

Because VB WithEvents fields are actually properties under the hood, Code Model needs to handle them specially. This was already done for "internal" code elements (i.e. elements defined in source) but wasn't done for "external" code elements. This change rectifies that.

Additionally, there was no good unit test infrastructure for testing external code model elements. This change adds that infrastructure plus new tests for both C# and VB.

**Testing Done**:
* Verified customer scenario
* Added unit test coverage specific to VB WithEvents fields in "external" code elements

Tagging @dotnet/roslyn-ide

### CODE REVIEW

File: src/VisualStudio/VisualBasic/Impl/CodeModel/VisualBasicCodeModelService.vb [hunk 1]
```diff
@@ -1341,20 +1341,62 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.CodeModel
...
             Dim access As EnvDTE.vsCMAccess = 0
 
-            ' TODO: Add vsCMAccessWithEvents for fields symbols defined as WithEvents
             Select Case symbol.DeclaredAccessibility
                 Case Accessibility.Private
                     access = access Or EnvDTE.vsCMAccess.vsCMAccessPrivate
                 Case Accessibility.Protected
                     access = access Or EnvDTE.vsCMAccess.vsCMAccessProtected
-                Case Accessibility.Internal
+                Case Accessibility.Internal, Accessibility.Friend
```
Comment (davkean) [inline src/VisualStudio/VisualBasic/Impl/CodeModel/VisualBasicCodeModelService.vb]: What's the difference between these?

**Comment by davkean** [comment]
:+1:

File: src/VisualStudio/VisualBasic/Impl/CodeModel/VisualBasicCodeModelService.vb (hunk 1 above)
Comment (DustinCampbell) [inline src/VisualStudio/VisualBasic/Impl/CodeModel/VisualBasicCodeModelService.vb]: There really isn't one. They have the same enum value.

**Comment by DustinCampbell** [comment]
@Pilchie, @jasonmalinowski or anyone else in @dotnet/roslyn-ide want to take a look?

**Comment by Pilchie** [comment]
:+1:  Love the new test helpers.

**Pull request merged**